packages/cactus-plugin-ledger-connector-xdai/src/test/typescript/integration/openapi/openapi-validation.test.ts

```diff
@@ -0,0 +1,405 @@
+  const apiClient = new XdaiApi(config);
+
+  await connector.getOrCreateWebServices();
+  await connector.registerWebServices(expressApp);
+
+  const fDeploy = "apiV1QuorumDeployContractSolidityBytecode";
+  const fInvoke = "apiV1QuorumInvokeContract";
+  const fRun = "apiV1QuorumRunTransaction";
+  const cOk = "without bad request error";
+  const cWithoutParams = "not sending all required parameters";
+  const cInvalidParams = "sending invalid parameters";
+
+  test(`${testCase} - ${fDeploy} - ${cOk}`, async (t2: Test) => {
+    const parameters = {
+      keychainId: keychainPlugin.getKeychainId(),
+      contractName: HelloWorldContractJson.contractName,
+      contractAbi: HelloWorldContractJson.abi,
+      constructorArgs: [],
+      web3SigningCredential: {
+        ethAccount: whalePubKey,
+        secret: whalePrivKey,
+        type: Web3SigningCredentialType.PrivateKeyHex,
+      },
+      bytecode: HelloWorldContractJson.bytecode,
+      gas: 1000000,
+    };
+    const res = await apiClient.deployContractV1(parameters);
+    t2.ok(res, "Contract deployed successfully");
+    t2.ok(res.data);
+    t2.equal(
+      res.status,
+      200,
+      `Endpoint ${fDeploy} in test case '${cOk}': status === 200`,
+    );
+
+    t2.end();
+  });
+
+  test(`${testCase} - ${fDeploy} - ${cWithoutParams}`, async (t2: Test) => {
+    try {
+      const parameters = {
+        contractName: HelloWorldContractJson.contractName,
+        contractAbi: HelloWorldContractJson.abi,
+        constructorArgs: [],
+        web3SigningCredential: {
+          ethAccount: whalePubKey,
+          secret: whalePrivKey,
+          type: Web3SigningCredentialType.PrivateKeyHex,
+        },
+        bytecode: HelloWorldContractJson.bytecode,
+        gas: 1000000,
+      };
+      await apiClient.deployContractV1(
+        (parameters as any) as DeployContractV1Request,
+      );
+    } catch (e) {
+      t2.equal(e.response.status, 400, "Bad request");
+      const fields = e.response.data.map((param: any) =>
+        param.path.replace(".body.", ""),
+      );
+      t2.ok(
+        fields.includes("keychainId"),
+        "Rejected because keychainId is required",
+      );
+    }
+
+    t2.end();
+  });
+
+  test(`${testCase} - ${fDeploy} - ${cInvalidParams}`, async (t2: Test) => {
+    try {
+      const parameters = {
+        keychainId: keychainPlugin.getKeychainId(),
+        contractName: HelloWorldContractJson.contractName,
+        contractAbi: HelloWorldContractJson.abi,
+        constructorArgs: [],
+        web3SigningCredential: {
+          ethAccount: whalePubKey,
+          secret: whalePrivKey,
+          type: Web3SigningCredentialType.PrivateKeyHex,
+        },
+        bytecode: HelloWorldContractJson.bytecode,
+        gas: 1000000,
+        fake: 4,
+      };
+      await apiClient.deployContractV1(
+        (parameters as any) as DeployContractV1Request,
+      );
+    } catch (e) {
+      t2.equal(e.response.status, 400, "Bad request");
```

Comment:
@elenaizaguirre The success message should say something like "fake 4 response.status === 400 OK"
This way if it fails in the error log we can exactly tell which assertion failed.